addOns/jwt/src/main/java/org/zaproxy/zap/extension/jwt/attacks/GenericAsyncTaskExecutor.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2020 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.jwt.attacks;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+import org.apache.log4j.Logger;
+import org.zaproxy.zap.extension.jwt.JWTActiveScanner;
+import org.zaproxy.zap.extension.jwt.JWTConfiguration;
+
+/**
+ * @author preetkaran20@gmail.com KSASAN
+ * @since TODO add version
+ * @param <T>
+ */
+public class GenericAsyncTaskExecutor<T> {
+
+    private Predicate<T> predicate;
+    private Iterator<T> iterator;
+    private ExecutorService executorService;
```

Comment:
Maintaining my own Executor Service based on the thread count supplied by the User in JWT settings.